src/main/java/edu/unc/lib/boxc/migration/cdm/services/CdmListIdService.java

```diff
@@ -0,0 +1,185 @@
+ * Service for listing CDM object IDs
+ * @author krwong
+ */
+
+public class CdmListIdService {
+    private static final Logger log = getLogger(CdmListIdService.class);
+
+    private CloseableHttpClient httpClient;
+    private String cdmBaseUri;
+
+    private int pageSize = 1000; //set default page size
+
+    /**
+     * Get the total number of object IDs for the given collection
+     * @param project
+     * @throw IOException
+     * @return
+     */
+    private int getTotalObjects(MigrationProject project) throws IOException {
+        String collectionId = project.getProjectProperties().getCdmCollectionId();
+        String totalObjectsUrl = "dmwebservices/index.php?q=dmQuery/" + collectionId + "/0/dmrecord/dmrecord/1/0/1/0/0/0/0/json";
+        String totalUri = URIUtil.join(cdmBaseUri, totalObjectsUrl);
+
+        ObjectMapper mapper = new ObjectMapper();
+        HttpGet getMethod = new HttpGet(totalUri);
+        try (CloseableHttpResponse resp = httpClient.execute(getMethod)) {
+            //Error looking up collection
+            String body = IOUtils.toString(resp.getEntity().getContent(), ISO_8859_1);
+            if (body.contains("Error looking up collection")) {
+                throw new MigrationException("No collection with ID '" + collectionId
+                        + "' found on server at " + totalUri);
+            }
+            JsonParser parser = mapper.getFactory().createParser(body);
+            if (parser.nextToken() != JsonToken.START_OBJECT) {
+                throw new MigrationException("Unexpected response from URL " + totalUri
+                        + "\nIt must be a JSON object, please check the response.");
+            }
+                ObjectNode entryNode = mapper.readTree(parser);
+                JsonNode pager = entryNode.get("pager");
+                int total = pager.get("total").asInt();
+                return total;
+        } catch (JsonParseException e) {
+            throw new MigrationException("Failed to parse response from URL " + totalUri
+                    + ": " + e.getMessage());
+        }
+    }
+
+    /**
+     * Page through the results
+     * @param project
+     * @throw IOException
+     * @return
+     */
+    private List<String> pagingResults (MigrationProject project, int total) {
+        int totalRecords = total;
+        int pageSize = this.pageSize;
+        int maxPages = totalRecords / pageSize;
+
+        List<String> urls = new ArrayList<>();
+
+        for (int page = 0; page <= maxPages; page++) {
+            int recordNum = (page * pageSize) + 1;
+            String cdmPageUrl = "dmwebservices/index.php?q=dmQuery/" + project.getProjectProperties().getCdmCollectionId() + "/0/dmrecord/dmrecord/" + pageSize + "/" + recordNum + "/1/0/0/0/0/json";
+            String pageUrl = URIUtil.join(cdmBaseUri, cdmPageUrl);
+            urls.add(pageUrl);
+        }
+        return urls;
+    }
+
+    /**
+     * Parse json for object IDs
+     * @param project
```

Comment:
Looks like the javadoc is a little out of sync with the code, there should be a `@param url` here